pysb/export/sbml.py

```diff
@@ -37,51 +21,47 @@ def _print_Symbol(self, sym):
 
-def format_single_annotation(annotation):
-    return '''<rdf:li rdf:resource="%s" />''' % annotation.object
-
-def get_annotation_preamble(meta_id):
-    return indent('''
-      <annotation>
-          <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:bqbiol="http://biomodels.net/biology-qualifiers/" xmlns:bqmodel="http://biomodels.net/model-qualifiers/">
-              <rdf:Description rdf:about="#%s">
-      ''' % meta_id)
-
-def get_annotation_postamble():
-    return indent('''
-              </rdf:Description>
-          </rdf:RDF>
-      </annotation>
-      ''')
-
-def get_species_annotation(meta_id, cp):
-    output = get_annotation_preamble(meta_id)
-    model = cp.monomer_patterns[0].monomer.model()
-    if len(cp.monomer_patterns) == 1:
-        # single monomer
-        all_annotations = model.get_annotations(cp.monomer_patterns[0].monomer)
-        groups = itertools.groupby(all_annotations, lambda a: a.predicate)
-        for predicate, annotations in groups:
-            qualifier = 'bqbiol:%s' % predicate
-            output += indent('<%s>\n    <rdf:Bag>\n' % qualifier, 12)
-            for a in annotations:
-                output += indent(format_single_annotation(a) + '\n', 20)
-            output += indent('    </rdf:Bag>\n</%s>\n' % qualifier, 12)
+
+def _check(value):
+    """If 'value' is None, prints an error message constructed using
+    'message' and then exits with status code 1.  If 'value' is an integer,
+    it assumes it is a libSBML return status code.  If the code value is
+    LIBSBML_OPERATION_SUCCESS, returns without further action; if it is not,
+    prints an error message constructed using 'message' along with text from
+    libSBML explaining the meaning of the code, and exits with status code 1.
+    """
+    if value is None:
+        raise ValueError('LibSBML returned a null value')
+    elif type(value) is int:
+        if value == libsbml.LIBSBML_OPERATION_SUCCESS:
+            return
+        else:
+            err_msg = 'Error encountered converting to SBML. ' \
+                      + 'LibSBML returned error code ' + str(value) + ': "' \
+                      + libsbml.OperationReturnValue_toString(value).strip() + '"'
+            raise ValueError(err_msg)
     else:
-        # complex
-        monomers = set(mp.monomer for mp in cp.monomer_patterns)
-        annotations = [a for m in monomers for a in model.get_annotations(m)
-                       if a.predicate in ('is', 'hasPart')]
-        qualifier = 'bqbiol:hasPart'
-        output += indent('<%s>\n    <rdf:Bag>\n' % qualifier, 12)
-        for a in annotations:
-            output += indent(format_single_annotation(a) + '\n', 20)
-        output += indent('    </rdf:Bag>\n</%s>\n' % qualifier, 12)
-    output += get_annotation_postamble()
-    return indent(output, 16)
+        return
+
+
+def _mathml_expr_call(expr, as_ast=False):
+    mathstr = '<apply><ci>{}</ci>'.format(expr.name)
```

Comment:
I'm nervous about building XML via string concatenation. `lxml.objectify` is great for succinctly building a DOM, although it would add a new dependency. It would look something like this:
```python
import lxml.objectify, lxml.etree

# Could construct these as globals
MATHML_NS = 'http://www.w3.org/1998/Math/MathML'
E = objectify.ElementMaker(annotate=False)
E_mathml = objectify.ElementMaker(
    namespace=MATHML_NS, nsmap={None: MATHML_NS}, annotate=False
)

# Function body
apply = E.apply()
apply.append(E.ci(expr.name))
# or if you like it on one line: apply = E.apply(E.ci(expr.name))
for sym in ...:
    apply.append(E.ci(sym.name ...))
if as_ast:
    math = E_mathml.math(apply)
    mathstr = lxml.etree.tostring(math)
    ...
else:
    return lxml.etree.tostring(apply)
```
Using the built-in `xml.dom.minidom` (as sympy's MathML printer does) probably wouldn't be that much more verbose, but I haven't worked with it much to say for sure.